app/models/concerns/host_common.rb

```diff
@@ -125,6 +125,9 @@ def crypt_root_pass
       else
         self.root_pass = operatingsystem.nil? ? PasswordCrypt.passw_crypt(unencrypted_pass) : PasswordCrypt.passw_crypt(unencrypted_pass, operatingsystem.password_hash)
         self.grub_pass = PasswordCrypt.grub2_passw_crypt(unencrypted_pass)
+        # correct encoding for SQLite3
+        self.root_pass = self.root_pass.force_encoding(Encoding::UTF_8) if self.root_pass.encoding == Encoding::ASCII_8BIT
```

Comment:
I see a couple of different encodings coming from `PasswordCrypt` depending on which crypt function's in use:

<pre>
2.0.0-p353 :010 > PasswordCrypt.passw_crypt('a').encoding
 => #<Encoding:ASCII-8BIT> 
2.0.0-p353 :011 > PasswordCrypt.passw_crypt('a', 'Base64').encoding
 => #<Encoding:US-ASCII> 
</pre>

So the conditional here probably is of limited use?  Or does the warning only happen with ASCII-8BIT and not US-ASCII?

You probably should put the assertion in both test paths, and consider moving this code to PasswordCrypt itself.